phantazm-commons/src/main/java/com/github/phantazmnetwork/commons/vector/BasicRegion3I.java

```diff
@@ -0,0 +1,30 @@
+package com.github.phantazmnetwork.commons.vector;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Standard implementation of {@link Region3I}. It is expected that only trusted code will construct instances of this
+ * record directly: the code must verify that the origin and lengths vectors comply with the general contract of
+ * Region3I.
+ * @param getOrigin the origin vector
+ * @param getLengths the lengths vector (all components are non-negative)
+ * @see Region3I
+ */
+record BasicRegion3I(@NotNull Vec3I getOrigin, @NotNull Vec3I getLengths) implements Region3I {
```

Comment:
nullcheck

Reply:
nullchecking is done implicitly in the factory methods in Region3I interface